api/v1alpha1/subscription_types.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2023 The Kubebb Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type InstallMethod string
+
+const (
+	// InstallMethodAuto means install directly without confirmation after detecting a new version.
+	InstallMethodAuto InstallMethod = "Auto"
+	// InstallMethodManual means installation process requires user's permission to proceed.
+	InstallMethodManual InstallMethod = "Manual"
+)
+
+// SubscriptionSpec defines the desired state of Subscription
+type SubscriptionSpec struct {
+	// ComponentRef is a reference to the Component
+	ComponentRef *corev1.ObjectReference `json:"component"`
+	// RepositoryRef is a reference to the Repository
+	RepositoryRef *corev1.ObjectReference `json:"repository"`
+	// ComponentPlanInstallMethod is the method used to install the component
+	ComponentPlanInstallMethod InstallMethod `json:"componentPlanInstallMethod,omitempty"`
+	// ForceInstallVersion represents the version that is forcibly installed by the user.
+	ForceInstallVersion string `json:"forceInstallVersion,omitempty"`
+}
+
+// SubscriptionStatus defines the observed state of Subscription
+type SubscriptionStatus struct {
+	ConditionedStatus `json:",inline"`
+	// HistoryVersions records the component version changes since this subscription created
+	// +optional
+	HistoryVersions []ComponetVersions `json:"historyVersions,omitempty"`
+
+	// Installed records all componentplans installed, ordered by install time.
+	// +optional
+	Installed []Installed `json:"installed,omitempty"`
```

Comment:
Let's keep `Subscription` simple which stores component version changes

Reply:
According to offline discussions, One `ComponentPlan` Represents one installation process, It may be auto **deleted** in a certain period of time (maybe 1 day or 1 week or 1 month) or when the number reaches a certain amount (similar to rs in deploy), So let us just track the **latest** one, and add label (maybe `kubebb.core.k8s.com.cn/subscription=<subscription-name>` ) to `ComponentPlan`.

Reply:
Looks good to me !